### PR TITLE
clemhirescal: Adds support for all MCP gain settings for filters A and D

### LINKED ISSUE #3190: clemhirescal: Add support for all MCP gain settings
**Description**  
`clemhirescal` is meant to radiometrically calibrate Clementine HIRES camera images to I/F by applying flat field correction and then scaling pixel values by a calibration constant, _k_, that is chosen from a lookup table based on the image filter and gain mode. The values of _k_ themselves have been empirically derived in order to scale I/F values of the HIRES camera to be consistent with the Clementine UVVIS camera. 

The k values currently used by `clemhirescal` are taken from  Tables 5 and 6 in  [Robinson et al. (2003)](https://doi.org/10.1029/2000JE001241), but the _k_ values in those tables only work for a tiny minority of the possible MCP gain states, and even then only apply to the A and D filters of the camera. The good news is that on page 11 of the paper, the authors state that _k_ values for arbitrary MCP gain states of the A and D filters can be derived by fitting a line to the values in Tables 5 and 6, respectively. 

Just doing a quick line fit in LibreOffice Calc, the calibration function for the A filter would be
`k_A = (-5.33333333333333*10^-5*MCP_gain) + 0.00937`

and for the D filter,
`k_D = (-9.75301204819275*10^-5*MCP_gain) + 0.0163866265`

(The paper also describes how a LUT of _k_ values can be established for the B and C filters, but this would require comparing a sample of HIRES images to co-registered UVVIS images and applying equations 5 and 6 from the paper [I don't know why the authors didn't just do this in the paper]. This would require considerable effort and it would only yield a first order approximation, so maybe not worth implementing at this time.)

**tl;dr**
`clemhirescal` currently uses a LUT of k values that only supports a minority of Clementine HIRES products, but the relevant source paper makes it sound like it's easy to derive k values to support all MCP gain states, at least for the A and D filters. Adding this feature would allow users to calibrate nearly all of the Clementine HIRES images from the A and D filters without necessarily having to derive calibration factors themselves.

**Acceptance Criteria**
- [ ] `clemhirescal` applies calibration function (as described on Page 11 of [Robinson et al. (2003)](https://doi.org/10.1029/2000JE001241)) to Clementine HIRES images taken through the A and D filters for arbitrary MCP gain states
- [ ] The updated version of `clemhirescal` must produce rasters with pixel values +/- 1% of results obtained with ISIS 3.8.0 for images collected in the subset of MCP gain states that are currently supported (no code regression)
- [ ] Update comments in the `clemhirescal` source code to correctly describe how the k values are determined in the updated code

**Example Data**
Processing consists of running `clem2isis` on Clementine HIRES EDRs (such as mentioned below), followed by `spiceinit` and then `clemhirescal`.
***A Filter***
An example image that *is* currently supported is `/pds_san/PDS_Archive/Clementine/cl_0051/lun224/lhxxxxxx/lhxxxxxe/lha0547e.224`. When this file is passed as input, an updated version of `clemhirescal` should yield a raster whose pixel values are each within 1% of a raster obtained by running the current version of `clemhirescal`.

An example image that is *not currently* supported is `/pds_san/PDS_Archive/Clementine/cl_0034/lun155/lhxxxxxx/lhxxxxxm/lha4988m.155`. This image should yield a calibrated output raster after `clemhirescal` is modified to use the calibration function described in [Robinson et al. (2003)](https://doi.org/10.1029/2000JE001241)

***D Filter***
An example image that *is* currently supported is `/pds_san/PDS_Archive/Clementine/cl_0054/lun237/lhxxxxxx/lhxxxxxn/lhd3924n.237`. When this file is passed as input, an updated version of `clemhirescal` should yield a raster whose pixel values are each within 1% of a raster obtained by running the current version of `clemhirescal`.

An example image that is *not currently* supported is `/pds_san/PDS_Archive/Clementine/cl_0018/lun088/lhxxxxxx/lhxxxxxh/lhd3062h.088`. This image should yield a calibrated output raster after `clemhirescal` is modified to use the calibration function described in [Robinson et al. (2003)](https://doi.org/10.1029/2000JE001241)

**Product Owner**
@dpmayerUSGS

### PR DESCRIPTION
**Pull request by AustinSanders**
## Description
Replaces the lookup tables in clemhirescal with calibration equations.  

## Related Issue
Resolves #3190 

## Motivation and Context
The current implementation scales pixel values by a calibration constant (k) that is selected from an incomplete lookup table.  Due to the incompleteness of the lookup tables, we are currently unable to support the full range of Clementine HIRES images unless the user manually derives and specifies a calibration factor.

## How Has This Been Tested?
One image from each filter type (A and D) was tested using the following command:
fx f1=1234_original_cal.cub f2=1234_cal.cub to=tolerance_boolean.cub equation="\(1+(0.01*f1))>=f2>=(1-(0.01*f1))"

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Review by scsides — COMMENTED** [review]
Do the equations match the hard coded values exactly? If there is even a slight difference we should document it in the XML. If not then the history comment should be enough.

File: isis/src/clementine/apps/clemhirescal/clemhirescal.xml [hunk 1]
```diff
@@ -9,14 +9,17 @@
       camera. It does so through the use of some constants based on offset and
```
Comment (dpmayerUSGS) [inline isis/src/clementine/apps/clemhirescal/clemhirescal.xml:9]: @AustinSanders Perhaps add some details to the description describing the (very small) differences in output radiances that result from switching from the look up tables to the modeled function?
This would address questions from @scsides  in his review.

The equation for the A filter yields identical k values that are identical to the ones in Table 5 from Robinson et al. (2003), therefore the pixel values in the output image are the same as the would be running the earlier code.

The equation for the D filter yields k values that differ from the hard-coded values by less than 1 standard deviation as reported in Table 6 of Robinson et al. (2003). The radiance values that result from computed k values differ by no more than 0.583% (that is, less than 1 percent) than radiance values computed from the hard-coded k values.

File: isis/src/clementine/apps/clemhirescal/clemhirescal.xml [hunk 2]
```diff
@@ -9,14 +9,17 @@
...
       flat images based on filter type. There are six possible offsets, with the
       resultant change being anywhere between eight and fifty to the DN. There
-      are four flat images, one for each of the four possible filters. 
-      Information for this program came from JGR Vol 108, A radiometric 
+      are four flat images, one for each of the four possible filters.
+      Information for this program came from JGR Vol 108, A radiometric
       calibration for the Clementine Hires camera: Robinson, Malaret, White.
```
Comment (dpmayerUSGS) [inline isis/src/clementine/apps/clemhirescal/clemhirescal.xml:14]: @AustinSanders Can you add the persistent link for the paper? https://doi.org/10.1029/2000JE001241

File: isis/src/clementine/apps/clemhirescal/clemhirescal.xml [hunk 3]
```diff
@@ -9,14 +9,17 @@
...
   </description>
   <history>
     <change name="Mackenzie Boyd" date="2009-08-03">
       Original version
     </change>
+    <change name="Austin Sanders" date="2019-11-06">
+      Replaced lookup tables with calibration equations.
```
Comment (dpmayerUSGS) [inline isis/src/clementine/apps/clemhirescal/clemhirescal.xml:34]: Add "See description for details." as per suggestion from @scsides

**Review by dpmayerUSGS — CHANGES_REQUESTED** [review]

File: isis/src/clementine/apps/clemhirescal/clemhirescal.xml (hunk 1 above)
Comment (AustinSanders) [reply isis/src/clementine/apps/clemhirescal/clemhirescal.xml:9]: @dpmayerUSGS Can you ensure that I properly captured the intended changes?

Comment (dpmayerUSGS) [reply isis/src/clementine/apps/clemhirescal/clemhirescal.xml:9]: @AustinSanders The changes to the documentation look good to me!

**Comment by jessemapel** [comment]
@dpmayerUSGS can you re-review indicating your requested changes have been met

**Comment by jessemapel** [comment]
@AustinSanders Do the tests need to be updated with this?

**Review by dpmayerUSGS — APPROVED** [review]

File: isis/src/clementine/apps/clemhirescal/clemhirescal.xml [hunk 4]
```diff
@@ -9,14 +9,31 @@
...
+      computed from the k values published in Table 6.
+
   </description>
   <history>
     <change name="Mackenzie Boyd" date="2009-08-03">
       Original version
     </change>
+    <change name="Austin Sanders" date="2019-11-06">
+      Replaced lookup tables with calibration equations.
+      See description for more detailed information.
```
Comment (jessemapel) [inline isis/src/clementine/apps/clemhirescal/clemhirescal.xml:35]: Add the github issue number to your history comment.

File: isis/src/clementine/apps/clemhirescal/main.cpp [hunk 5]
```diff
@@ -48,43 +48,25 @@ void IsisMain() {
...
-        abscoef = 0.001375;
-      }
-      else if(MCP == 158) {
-        abscoef = 0.00097;
-      }
-      else {
-        QString message = "Image is not one of supported MCP Gain Mode IDs, enter your own K value";
-        throw IException(IException::User, message, _FILEINFO_);
-      }
+        abscoef = ((-9.75301204819275 * pow(10, -5) * MCP) + 0.0163866265);
```
Comment (jessemapel) [inline isis/src/clementine/apps/clemhirescal/main.cpp:67]: Since this has changed do tests need to be updated? Do we have tests for both of these?

Comment (dpmayerUSGS) [reply isis/src/clementine/apps/clemhirescal/main.cpp:67]: @jessemapel @AustinSanders I don't think there *are* any tests of the output of `clemhirescal`.

Comment (AustinSanders) [reply isis/src/clementine/apps/clemhirescal/main.cpp:67]: I updated the test data, but I was instructed not to push it back to the test data location until the PR was merged.

**Comment by jessemapel** [comment]
See #3503
